lib/config.module.ts

```diff
@@ -125,6 +126,10 @@ export class ConfigModule {
   ): Record<string, any> {
     try {
       const envFilePath = options.envFilePath || resolve(process.cwd(), '.env');
+      if (options.expandVariables) {
+        const config = dotenv.config({ path: envFilePath });
```

Comment:
Can't we use `dotenv.parse` here? (same as below)

Reply:
@kamilmysliwiec The `dotenv-expand` package requires you to pass the config result from `dotenv` to the default function, which should be in the form:

```ts
{
  error: Error
  parsed: {
    [name: string]: string;
  }
}
```

`.parse()` returns `{ [name: string]: string; }` directly which causes the output/result of `dotenvExpand` to be `undefined`.

**dotenv-expand:**
![image](https://user-images.githubusercontent.com/7609893/72383017-829f1e00-36e8-11ea-9c03-22228da5828c.png)

**dotenv**
![image](https://user-images.githubusercontent.com/7609893/72383055-9d719280-36e8-11ea-8d50-206df8d2b72d.png)

Reply:
However, `dotenv.config()` automatically assigns variables to `process.env` which will cause side-effects (see this line https://github.com/nestjs/config/pull/40/files#diff-7bffff97f29773aca9dd0b43601ef512L142)

Reply:
@kamilmysliwiec Got it. Just pushed a fix. That should do it. 😄

Reply:
Thanks!